### PR TITLE
Gui: Add about image for dev version

### PR DESCRIPTION
**Pull request by qewer33**
Following the discussion with @adrianinsaval on my recent PR: https://github.com/FreeCAD/FreeCAD/pull/16548

This PR adds a different about dialog thumbnail image for development versions. This way we can have separate images for dev versions and releases.

@adrianinsaval I'd like to hear your opinion regarding this, will it pose any problems for branded versions? I haven't implemented any special fallback logic so you will need to add another about dialog image for Ondsel postfixed with "dev", it can be same image, it doesn't matter.

image

### CODE REVIEW

**Comment by maxwxyz** [comment]
IMO version number should be not in the image itself but as text overlay like in the splashscreen.

**Review by adrianinsaval — APPROVED** [review]

**Comment by adrianinsaval** [comment]
> IMO version number should be not in the image itself but as text overlay like in the splashscreen.

I don't think it's worth the effort, in any case we can skip the version number since it's displayed in the dialog anyway.

**Comment by kadet1090** [comment]
I'd skip the version altogether - there it is not really useful here as it is just below and it is yet another thing to be updated when version changes. What is worse there is no source files attached so it would need to be remade from scratch just to update the version.

**Comment by adrianinsaval** [comment]
good point, if this image is being generated from an svg I believed the svg could be used directly

**Comment by qewer33** [comment]
Currently these about banners are updated on a version by version basis anyway, "updating" the version numbers is trivial. But if there's consensus I can drop the version numbers. I personally find it nice to see the version right below the FreeCAD logo on the about dialog.

Here's Inkscape doing it creatively:

image

**Comment by kadet1090** [comment]
> "updating" the version numbers is trivial
As long as you have the source files, and even if trivial it is yet another thing to remember.

If we have it in form of SVG file we can even change it via code probably - as long as you use something like `${VERSION_NUMBER}` placeholder it should be quite trivial to find and replace in SVG file when loading. We do that for colors already.

**Comment by pierreporte** [comment]
You will need to embed the typeface to have correct rendering.

**Review by luzpaz — APPROVED** [review]

**Comment by maxwxyz** [comment]
Looked at this at the merge meeting but it still has conflicts. Could you please resolve them?

**Comment by maxwxyz** [comment]
@qewer33 ping for conflicts

**Comment by maxwxyz** [comment]
@qewer33 bump

**Comment by maxwxyz** [comment]
@qewer33 still conflicts

**Comment by chennes** [comment]
I think I have resolved the conflict correctly, but I need someone else to check it to ensure I didn't choose the wrong code path.

**Comment by maxwxyz** [comment]
@chennes not sure if your last commit made CI fail?

**Comment by chennes** [comment]
The problem is that this PR does not take into account the changes made in #16438 -- so it needs more than a simple merge or rebase, it needs to be reconfigured to use the newly-created DlgAbout.cpp file/class structure.

**Comment by maxwxyz** [comment]
marking it as stale for now until the author revisits this

**Comment by luzpaz** [comment]
@ein-shved any chance you could jump in here and resurrect this PR ? :sweat_smile:

**Comment by ein-shved** [comment]
> @ein-shved any chance you could jump in here and resurrect this PR ? 😅

Ok, will try to reproduce the build failure on my NixOS.

**Comment by luzpaz** [comment]
Chennes's provides orientation here: https://github.com/FreeCAD/FreeCAD/pull/17030#issuecomment-2545776095

**Comment by ein-shved** [comment]
image
I think, this is what was expected. Dev window is above, my system freecad's window is below.

**Comment by luzpaz** [comment]
Well played @ein-shved :pray:

File: src/Gui/MainWindow.cpp [hunk 1]
```diff
@@ -1855,10 +1855,52 @@ void MainWindow::startSplasher()
 
 void MainWindow::stopSplasher()
 {
-    if (d->splashscreen) {
-        d->splashscreen->finish(this);
-        delete d->splashscreen;
-        d->splashscreen = nullptr;
+    const auto isWayland = qGuiApp->platformName() == QLatin1String("wayland");
+    if (isWayland) {
```
[the remark this answers is not in the document]
Comment (3x380V) [reply src/Gui/MainWindow.cpp:1859]: @ein-shved, that comes from #16264 and was not fully reverted. Long story short, startup process needs some refactoring and this is remnant from shortcut to get splash screen displayed a bit longer. It didn't quite fly.

**Comment by ein-shved** [comment]
> Well played @ein-shved :pray: 

Welcome to review then) #18599

File: src/Gui/MainWindow.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ein-shved) [reply src/Gui/MainWindow.cpp:1859]: @3x380V got it, thank you!

**Comment by adrianinsaval** [comment]
closing this one in favor of #18599 then

**Pull request closed without merging**